hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -7,21 +7,212 @@
+            success of each operation is verified via backdoor.
             '''
       milestone: V1
       tests: ["flash_ctrl_sanity"]
     }
+    {
+      name: backpressure
+      desc: '''
+            Sanity + randomly pick between back-pressured reads / writes and fifo status-polled
+            reads / writes via the controller.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: all_partitions
+      desc: '''
+            Sanity + both, legal data and info partitions are accessed. In future, support for
+            multiple info partitions may be added - those will be covered as well.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_read_check
+      desc: '''
+            Sanity + the success of each operation is verified by randomly choosing
+            between backdoor and via the host interface.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_illegal_access
+      desc: '''
+            Do a bunch of illegal accesses via host interface to ensure that an error response is
+            appropriately seen. This includes writes, out of bound reads, protocol violated reads
+            and all other unsupported accesses.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_outstanding_access
+      desc: '''
+            Drive back-to-back access to ensure coverage on the maximum supported outstanding
+            on this interface.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: intr_enabled
+      desc: '''
+            Sanity + enable all interrupts and verify correctness of assertion and de-assertion of
+            each pin. Replace polling with waits on interrrupts.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_ctrl_arb
+      desc: '''
+            Continuously read data from host interface while randomly programming or reading from
+            the flash controller. Ensure the expected bandwidths are seen on both interfaces.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_read_collide
+      desc: '''
+            Read LSB bus word via host and program or erase the MSB bus word at the same time. Read
+            the MSB bus word within the same flash data word - it should read back the updated data,
+            proving that the internal buffer was cleared.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: en_mp_regions
+      desc: '''
+            Protect randomized and legal ranges of data and info partitions and ensure that the
+            corresponding read / program / erase enable bits are working. TBD - future support for
+            sub-page protection may be added - that will be tracked as a part of this test as well.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: program_erase_verify
```

Comment:
it's possible that this will come as part of the vendor flash directly. If that's the case, we may not need this at all.

Reply:
Oh, ok. If the verify fails, I imagine the eflash would to send back an error somehow? How will that be captured?

Reply:
i haven't seen the actual vendor doc so i can't say :)
But I think what should happen is that the macro would report a failure, and I will propagate this all the way back to the flash controller which should show up in the status.

Reply:
Removed it for now.